sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -230,53 +225,46 @@ private HttpResponseLoggingContext getResponseLoggingOptions(HttpResponse httpRe
     private final class DefaultHttpRequestLogger implements HttpRequestLogger {
         @Override
         public Mono<Void> logRequest(ClientLogger logger, HttpRequestLoggingContext loggingOptions) {
-            final LogLevel logLevel = getLogLevel(loggingOptions);
-            if (logger.canLogAtLevel(logLevel)) {
-                log(logLevel, logger, loggingOptions);
-            }
-
+            logRequestSync(logger, loggingOptions);
             return Mono.empty();
         }
 
         @Override
         public void logRequestSync(ClientLogger logger, HttpRequestLoggingContext loggingOptions) {
-            final LogLevel logLevel = getLogLevel(loggingOptions);
-            if (logger.canLogAtLevel(logLevel)) {
-                log(logLevel, logger, loggingOptions);
-            }
+            log(getLogLevel(loggingOptions), logger, loggingOptions);
         }
 
         private void log(LogLevel logLevel, ClientLogger logger, HttpRequestLoggingContext loggingOptions) {
-            final HttpRequest request = loggingOptions.getHttpRequest();
-            LoggingEventBuilder logBuilder = getLogBuilder(logLevel, logger);
+            if (!logger.canLogAtLevel(logLevel) || httpLogDetailLevel == HttpLogDetailLevel.NONE) {
```

Comment:
TODO: this might not be right - we probably logged something at level None

Reply:
Nothing is logged when the HttpLogDetailLevel is `NONE`, `process` and `processSync` effectively no-op in that case.